lnd.go

```diff
@@ -973,7 +973,16 @@ func startRestProxy(cfg *Config, rpcServer *rpcServer, restDialOpts []grpc.DialO
 			},
 		},
 	)
-	mux := proxy.NewServeMux(customMarshalerOption)
+	mux := proxy.NewServeMux(
+		customMarshalerOption,
+
+		// Don't allow falling back to other HTTP methods, we want exact
+		// matches only. The actual method to be used can be overwritten
+		// by setting X-HTTP-Method-Override so there should be no
+		// reason for not specifying the correct method in the first
```

Comment:
Oh wow, surprising that this is _default_ behavior!

Reply:
Curious if there're any sort of accidentally-working interactions deployed in the wild that incorrectly relied on this behavior. Either way, this is definitely a bug w.r.t _our_ advertised interface.